### PR TITLE
Clean ups and CPU/CODEC DAI mapping improvements

### PR DESCRIPTION
**Pull request by charleskeepax**
Some more minor clean up to the code and update the mapping between the CPU and CODEC DAIs within the DAI link to use the actual mapping.

### CODE REVIEW

File: sound/soc/intel/boards/sof_sdw.c [hunk 1]
```diff
@@ -1568,11 +1563,6 @@ static int create_sdw_dailink(struct snd_soc_card *card, int *link_index,
 			return -EINVAL;
 		}
 
-		sdw_codec_ch_maps = devm_kcalloc(dev, codec_num,
-						 sizeof(*sdw_codec_ch_maps), GFP_KERNEL);
-		if (!sdw_codec_ch_maps)
-			return -ENOMEM;
-
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c:1586]: Can we really use a global ch_maps?

Before we allocated one ch_maps per stream/direction, now we have the same for capture and playback.

I wonder if we have any asymmetric configurations where we'd need a different mapping?

Comment (charleskeepax) [reply sound/soc/intel/boards/sof_sdw.c:1586]: It is quite plausible that one could create a system where the mapping might be different for capture and playback. However, that is completely not supported by the current code, true it does creates two copies of the table but:

 a) they are both based on the same CPU and CODEC data so will always be identical.
 b) the current snd_soc_acpi_* structures are incapable of expressing such a topology as well, since direction is a part of the codec_info rather than those structs so an endpoint is currently assumed to be bidirectional.
 c) the code doesn't properly support multiple endpoints on one device yet either, which would be needed to group different devices into playback and capture endpoints too.

So in short I would agree it is possible, but significant other refactoring is necessary to support it and I would argue that such refactoring would likely change a lot of this code anyway. I am working through all of these items slowly, but this patch fixes an issue where the mapping is incorrect without needing to refactor lots of other bits and add other features first.

Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c:1586]: fair enough, thanks @charleskeepax

**Review by plbossart — APPROVED** [review]

**Review by bardliao — APPROVED** [review]
